Build/libHttpClient.Android/src/main/java/com/xbox/httpclient/HttpClientRequest.java

```diff
@@ -74,6 +92,44 @@ public void onResponse(Call call, final Response response) {
         });
     }
 
+    private String GetAllNetworksInfo() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            return "API version too old - unable to get network info!";
+        }
+
+        ConnectivityManager cm = (ConnectivityManager)appContext.getSystemService(Context.CONNECTIVITY_SERVICE);
```

Comment:
is this to check if it's currently connected? Or what the current HTTP service is?

Reply:
Yeah, the point of this PR is to gather and log as much info as we can about the state of the various networks the device is connected to. The `ConnectivityManager` is the Android entrypoint to that information, and this function is just designed to collect that info and marshal it into a string we can log.